### PR TITLE
fix(database_observability.mysql): Exclude system schemas from MySQL health check

### PR DESCRIPTION
**Pull request by matthewnolf**
### Brief description of Pull Request
Include excluded schemas in the health check that confirms performance_schema has data. This refines the check to verify that application queries are present, and not just system queries that populate even on idle databases.

### PR Checklist

- [ ] Documentation added
- [ ] Tests updated
- [ ] Config converters updated

### CODE REVIEW

File: internal/component/database_observability/mysql/collector/health_check.go [hunk 1]
```diff
@@ -221,10 +224,11 @@ func checkRequiredGrants(ctx context.Context, db *sql.DB) healthCheckResult {
...
 }
 
-// checkEventsStatementsDigestHasRows ensures performance_schema.events_statements_summary_by_digest has rows.
-func checkEventsStatementsDigestHasRows(ctx context.Context, db *sql.DB) healthCheckResult {
+// checkEventsStatementsDigestHasRows ensures performance_schema.events_statements_summary_by_digest has rows,
+// excluding system schemas.
+func (c *HealthCheck) checkEventsStatementsDigestHasRows(ctx context.Context, db *sql.DB) healthCheckResult {
 	r := healthCheckResult{name: "PerformanceSchemaHasRows"}
-	const q = `SELECT COUNT(*) FROM performance_schema.events_statements_summary_by_digest`
+	q := fmt.Sprintf(`SELECT COUNT(*) FROM performance_schema.events_statements_summary_by_digest WHERE schema_name NOT IN %s`, buildExcludedSchemasClause(c.excludeSchemas))
```
Comment (ghost) [inline internal/component/database_observability/mysql/collector/health_check.go:231]: **Semgrep identified a blocking 🔴 issue in your code:**
String-formatted SQL query detected. This could lead to SQL injection if the string is not sanitized properly. Audit this call to ensure the SQL is not manipulable by external data.

Why this might be safe to ignore:
>The SQL query uses buildExcludedSchemasClause which appears to be a helper function that constructs a safe SQL clause from the excludeSchemas configuration. Since this is configuration-driven rather than user-supplied input, and the function name suggests it properly formats the schema list, this is not exploitable for SQL injection.

**To resolve this comment:**



🔧 No guidance has been designated for this issue. Fix according to your organization's approved methods.




<details>
<summary>💬 Ignore this finding</summary>
<br>


Reply with Semgrep commands to ignore this finding.
- `/fp <comment>` for false positive
- `/ar <comment>` for acceptable risk
- `/other <comment>` for all other reasons


Alternatively, triage in [Semgrep AppSec Platform](https://semgrep.dev/orgs/grafana_labs/findings/763442837) to ignore the finding created by [string-formatted-query](https://semgrep.dev/playground/r/ZRTKA2q/go.lang.security.audit.database.string-formatted-query.string-formatted-query?utm_campaign=finding_notification&utm_medium=review_comment&utm_source=github&utm_content=rule).

</details>



We're currently testing semgrep's diff-aware PR comment feature on a subset of our repos-- if you run into issues or find this spammy, please reach out to @danny.cooper in slack and give feedback. 

For backwards compatability with gosec, its best to use polyglot suppression comments of the following format for false positives:
`// #nosec <gosec rule ID> nosemgrep: <semgrep rule ID>`


<sub>*You can view more details about [this finding](https://semgrep.dev/orgs/grafana_labs/findings/763442837) in the Semgrep AppSec Platform.*</sub>

File: internal/component/database_observability/mysql/collector/health_check_test.go [hunk 2]
```diff
@@ -33,6 +33,7 @@ func TestHealthCheck(t *testing.T) {
 		collector, err := NewHealthCheck(HealthCheckArguments{
 			DB:              db,
 			CollectInterval: 100 * time.Millisecond,
+			ExcludeSchemas:  nil,
```
Comment (cristiangreco) [inline internal/component/database_observability/mysql/collector/health_check_test.go:36]: Nit: can we add a test with non-empty `ExcludeSchemas` as well?

Comment (matthewnolf) [reply internal/component/database_observability/mysql/collector/health_check_test.go:36]: Done in 2cb7af442b4a2fccbaa3efa80733f05f3aea205b

**Review by cristiangreco — APPROVED** [review]
